lib/Client/Util.php

```diff
@@ -26,9 +26,31 @@ public function randInt32() {
         return $this->_rng->int32();
     }
 
-    public function nowMicros() {
+    public static function nowMicros() {
         // Note: microtime returns the current time *in seconds* but with
         // microsecond accuracy (not the current time in microseconds!).
         return floor(microtime(TRUE) * 1000.0 * 1000.0);
     }
+
+    /**
+     * Add an item to the array, ensuring that the size of the array never exceed the max. If adding the item would
+     * cause the array count to exceed max, then the item is not added.
+     *
+     * @param array $arr Array to add item to
+     * @param object $item Item to be added
+     * @param int $max Max number of items allowed in array, if provided max is <= 0, 1 will be used as the max
+     * @return bool True if the item was successfully added to the array
+     */
+    public function pushIfSpaceAllows(&$arr, $item, $max) {
```

Comment:
This is so much better :) We could optionally make it static since it doesn't use the instance variable anymore.